toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -249,35 +201,156 @@ func (b *LiveOSIsoBuilder) prepareRootfsForDracut(writeableRootfsDir string) err
 	return nil
 }
 
-// prepareLiveOSDir
+func (b *LiveOSIsoBuilder) updateGrubCfg(grubCfgFileName string, extraCommandLine string) error {
+
+	inputContentString, err := file.Read(grubCfgFileName)
+	if err != nil {
+		return err
+	}
+
+	searchCommand := fmt.Sprintf(searchCommandTemplate, isoMediaLabel)
+	rootValue := fmt.Sprintf(rootValueTemplate, isoMediaLabel)
+
+	inputContentString, err = setSearchCommand(inputContentString, searchCommand)
+	if err != nil {
+		return fmt.Errorf("failed to update the search command in the iso grub.cfg:\n%w", err)
+	}
+
+	inputContentString, oldLinuxPath, err := setLinuxPath(inputContentString, isoKernelPath)
+	if err != nil {
+		return fmt.Errorf("failed to update the kernel file path in the iso grub.cfg:\n%w", err)
+	}
+
+	inputContentString, err = replaceToken(inputContentString, oldLinuxPath, isoKernelPath)
```

Comment:
Why would there be more occurrences of this path in the grub config file?

Reply:
The grub.cfg I have been looking at had multiple occurrences of initrd.img path. There's always a chance that a file is authored with references to the kernel path - so, if we're updating it, we should probably update all occurrences.

The initrd.img case was kind of silly though. I'm not sure why they have the check if the image layout is determined at build time - and if they have no fallback. But either way, from our end, we need to keep the path consistent at least across this file.
```if -f initrd.img
  initrd initrd.img
fi```